### ISSUE TITLE
Scan crash for very strange case

### ISSUE THREAD

**Issue by nouiz** [Theano/Theano#3764]
In https://github.com/Theano/Theano/pull/3580 a tests was added for a corner case of shape inference. In https://github.com/Theano/Theano/pull/3709, we reverted gh-3580 as it need more investigation, but the new strange test that was added have the execution of scan commented as the execution of the scan.

So uncomment that execution and fix it. If we reimplement the infer_shape, we must return the output of the scan, not just the shape, as this will remove the execution, so it would remove the error.

**Comment by Sentient07** [comment]
@nouiz , can i take this issue ?

**Comment by nouiz** [comment]
Nobody is working on it, but learning scan internal is very hard and we won't be very active for the next 2 weeks due to holidays. So if you want to work on something in the next 2 weeks, I would suggest something else.

I updated a little the description to help understand what is needed todo.

**Comment by Sentient07** [comment]
Yeah, i want to work on something in the next two weeks. Could you please assign me something ? :) 
Also, i'd try to understand this from your updated comment

**Comment by nouiz** [comment]
I think you should forget this one for now. The internal will be too complicated without help. I found one other issue. I'll find a second one.

**Comment by Sentient07** [comment]
@nouiz yeah, will wait for your holidays to get over. Have a nice time. Thanks!
